### PR TITLE
[Storage] `az storage blob query`: Support Azure Storage Query Acceleration

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

Fix issue #14323 
Add data plane support for Query Acceleration feature (https://docs.microsoft.com/azure/storage/blobs/data-lake-storage-query-acceleration) into CLI.

**Testing Guide**  

- Prepare blob named `quick_query.csv`
    - Show result directly
    `az storage blob query -c test -n quick_query.csv --query-expression "SELECT _2 FROM BlobStorage"  --account-name zuhdefault --account-key xxxxxx`
    - Save result in file
     `az storage blob query -c test -n quick_query.csv --query-expression "SELECT _2 FROM BlobStorage"  --account-name zuhdefault --account-key xxxxxx --result-file xxx.csv`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -814,6 +814,11 @@
 short-summary: Manage blob metadata.
 """
 
+helps['storage blob query'] = """
+type: command
+short-summary: Enable users to select/project on blob/or blob snapshot data by providing simple query expressions.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:819]: `select/project on blob/or blob snapshot` This means "select project on blob or blob snapshot", right? What does `/` mean?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -819,6 +819,84 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
         c.argument('source_lease_id', arg_group='Copy Source')
 
+    with self.argument_context('storage blob query') as c:
+        from ._validators import validate_text_configuration
+        c.register_blob_arguments()
+        c.register_precondition_options()
+        line_separator = CLIArgumentType(help="The string used to separate records.", default='\n')
+        column_separator = CLIArgumentType(help="The string used to separate columns.", default=',')
+        quote_char = CLIArgumentType(help="The string used to quote a specific field.", default='"')
+        record_separator = CLIArgumentType(help="The string used to separate records.", default='\n')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:829]: How about the default values of these arguments also be prompted in help?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -819,6 +819,84 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+        c.argument('quote_char', arg_group='Delimited CSV Configuration',
+                   help="The string used to quote a specific field. Default to '\"'.")
+        c.argument('record_separator', arg_group='Delimited CSV Configuration',
+                   help="The string used to separate records. Default to '\\n'")
+        c.argument('escape_char', arg_group='Delimited CSV Configuration',
+                   help='The string used as an escape character. Default to empty.')
+        c.argument('has_header', arg_group='Delimited CSV Configuration', arg_type=get_three_state_flag(),
+                   help='Whether the blob data includes headers in the first line. The default value is False, '
+                   'meaning that the data will be returned inclusive of the first line. If set to True, the data '
+                   'will be returned exclusive of the first line.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:898]: > line_separator = CLIArgumentType(help="The string used to separate records.", default='\n')
column_separator = CLIArgumentType(help="The string used to separate columns.", default=',')
quote_char = CLIArgumentType(help="The string used to quote a specific field.", default='"')
record_separator = CLIArgumentType(help="The string used to separate records.", default='\n')
escape_char = CLIArgumentType(help="The string used as an escape character. Default to empty.", default="")

Can these CLIArgumentTypes be reused here?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: removed this part

File: src/azure-cli/azure/cli/command_modules/storage/_help.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py]: updated

**Review by zhoxing-ms — APPROVED** [review]
LGTM

**Pull request merged**